corehq/apps/geospatial/static/geospatial/js/models.js

```diff
@@ -0,0 +1,542 @@
+hqDefine('geospatial/js/models', [
+    'jquery',
+    'knockout',
+    'hqwebapp/js/initial_page_data',
+], function (
+    $,
+    ko,
+    initialPageData
+) {
+    const HOVER_DELAY = 400;
+    const DOWNPLAY_OPACITY = 0.2;
+    const FEATURE_QUERY_PARAM = 'features';
+    const DEFAULT_CENTER_COORD = [-20.0, -0.0];
+
+    var MissingGPSModel = function () {
+        this.casesWithoutGPS = ko.observable([]);
+        this.usersWithoutGPS = ko.observable([]);
+    };
+
+    var SavedPolygon = function (polygon) {
+        var self = this;
+        self.text = polygon.name;
+        self.id = polygon.id;
+        self.geoJson = polygon.geo_json;
+    };
+
+    var MapItem = function (itemId, itemData, marker, markerColors) {
+        'use strict';
+        var self = this;
+        self.itemId = itemId;
+        self.itemData = itemData;
+        self.marker = marker;
+        self.selectCssId = "select" + itemId;
+        self.isSelected = ko.observable(false);
+        self.markerColors = markerColors;
+
+        self.groupId = null;
+        self.groupCoordinates = null;
+
+        self.setMarkerOpacity = function (opacity) {
+            let element = self.marker.getElement();
+            element.style.opacity = opacity;
+        };
+
+        function changeMarkerColor(selectedCase, newColor) {
+            let marker = selectedCase.marker;
+            let element = marker.getElement();
+            let svg = element.getElementsByTagName("svg")[0];
+            let path = svg.getElementsByTagName("path")[0];
+            path.setAttribute("fill", newColor);
+        }
+
+        self.getItemType = function () {
+            if (self.itemData.type === "user") {
+                return gettext("Mobile Worker");
+            }
+            return gettext("Case");
+        };
+
+        self.isSelected.subscribe(function () {
+            var color = self.isSelected() ? self.markerColors.selected : self.markerColors.default;
+            changeMarkerColor(self, color);
+        });
+    };
+
+    var ClusterMapItem = function (itemId, itemData, link) {
```

Comment:
Is there a better name for this model? Its not really clear what this is.

Reply:
This is similar to the `mapItem` class, however the difference is that this gets used as the case data class for maps that use the grouping/clustering functionality. Would a comment here explaining this difference suffice? Otherwise I'm happy to rename it, perhaps to something like `GroupedCaseMapItem`.

Reply:
Not sure what the best name would be, though, GroupedCasesMapItem sounds good